sklearn/metrics/tests/test_pairwise_distances_reduction.py

```diff
@@ -0,0 +1,466 @@
+    )
+
+    ref_dist, ref_indices = PairwiseDistancesReduction.get_for(
+        X, Y, parameter, metric="fast_euclidean"
+    ).compute(return_distance=True)
+
+    dist, indices = PairwiseDistancesReduction.get_for(
+        X, Y, parameter, metric="fast_euclidean", n_threads=1
+    ).compute(return_distance=True)
+
+    ASSERT_RESULT[PairwiseDistancesReduction](ref_dist, dist, ref_indices, indices)
+
+
+@pytest.mark.parametrize("seed", range(5))
+@pytest.mark.parametrize("n_samples", [100, 1000])
+@pytest.mark.parametrize("metric", PairwiseDistancesReduction.valid_metrics())
+@pytest.mark.parametrize(
+    "PairwiseDistancesReduction",
+    [PairwiseDistancesArgKmin],
+)
+def test_strategies_consistency(
+    PairwiseDistancesReduction,
+    metric,
+    n_samples,
+    seed,
+    n_features=10,
+    dtype=np.float64,
+):
+    # Results obtained using both parallelization strategies must be identical
+    if _in_unstable_openblas_configuration() and metric == {
+        "fast_sqeuclidean",
+        "fast_euclidean",
+    }:
+        pytest.xfail(
+            "OpenBLAS (used for 'fast_(sq)euclidean') is unstable in this configuration"
+        )
+
+    rng = np.random.RandomState(seed)
+    spread = 100
+    X = rng.rand(n_samples, n_features).astype(dtype) * spread
+    Y = rng.rand(n_samples, n_features).astype(dtype) * spread
+
+    # Haversine distance only accepts 2D data
+    if metric == "haversine":
+        X = np.ascontiguousarray(X[:, :2])
+        Y = np.ascontiguousarray(Y[:, :2])
+
+    parameter = (
+        10
+        if PairwiseDistancesReduction is PairwiseDistancesArgKmin
+        # Scaling the radius slightly with the numbers of dimensions
+        else 10 ** np.log(n_features)
+    )
+
+    pairwise_distances_reduction = PairwiseDistancesReduction.get_for(
+        X,
+        Y,
+        parameter,
+        metric=metric,
+        # Taking the first
+        metric_kwargs=_get_dummy_metric_params_list(metric, n_features)[0],
+        # To be sure to use parallelization
+        chunk_size=n_samples // 4,
+    )
+
+    dist_par_X, indices_par_X = pairwise_distances_reduction.compute(
+        strategy="parallel_on_X", return_distance=True
+    )
+
+    dist_par_Y, indices_par_Y = pairwise_distances_reduction.compute(
+        strategy="parallel_on_Y", return_distance=True
+    )
+
+    ASSERT_RESULT[PairwiseDistancesReduction](
+        dist_par_X, dist_par_Y, indices_par_X, indices_par_Y
+    )
+
+
+@fails_if_unstable_openblas
+@pytest.mark.parametrize("seed", range(10))
+@pytest.mark.parametrize("n_samples", [100, 1000])
+@pytest.mark.parametrize("n_features", [5, 10, 100])
+@pytest.mark.parametrize("k, radius", [(50, 100)])
+def test_fast_sqeuclidean_correctness(
+    seed,
+    n_samples,
+    n_features,
+    k,
+    radius,
+    dtype=np.float64,
+):
+    # The fast squared euclidean strategy must return results
+    # that are close to the ones obtained with the euclidean distance
+    if n_samples < k:
```

Comment:
I think we can remove this, because `n_samples < k` is always false for all test parameters.